### PR TITLE
fix: avoid race condition by not running exec/start twice

### PR DESCRIPTION
**Pull request by ZauberNerd**
ContainerExecAttach implicitly runs ContainerExecStart while attaching
to stdout/stderr.
Ref: https://github.com/moby/moby/blob/e02bc91dcbf6ab70ab39352f3577a4394b0f863a/client/container_exec.go#L40

Calling both can lead to a race condition as observed in #627

Fixes: #627

### CODE REVIEW

**Review by catthehacker — DISMISSED** [review]

**Review by cplee — DISMISSED** [review]

**Comment by cplee** [comment]
@ZauberNerd - i need you to rebase this PR before it can be merged...or allow reviewers to update the branch.

**Review by catthehacker — APPROVED** [review]

**Comment by ZauberNerd** [comment]
> @ZauberNerd - i need you to rebase this PR before it can be merged...or allow reviewers to update the branch.

I've rebased the branch. I can't see the "Allow maintainers to edit the PR" checkbox. Maybe it doesn't work for organisation accounts?

**Review by cplee — APPROVED** [review]

**Pull request merged**
